types/i18n.d.ts

```diff
@@ -1,20 +1,21 @@
 export declare class I18n {
   t (key: string, templateData: Object, options: Object): string
   add (items: Object, locale: string, prefix: string): I18n
-  setLocale (locale: string): I18n
+  setLocale (locale: string | null | undefined): I18n
```

Comment:
I would recommend setting the type without the null:

```suggestion
setLocale (locale: string | undefined): I18n
```

Reply:
But I'm no TS expert, but my reasoning behind including `null` is that DOM methods often return `null` rather than `undefined`, so expressions like the following can be written without checks or conversions:

```js
i18n.setLocale(document.getElementById('my-elem').getAttribute('lang'))
```

Reply:
You are right, DOM checks do return `null` values so it will need to be declared. I'll resolve this so please ignore the comment.